Submodule src/3rdparty cc3e3122a..3302d1ad6:
diff --git a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
index 66ed8434ff052..1bc73070d395e 100644
--- a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
+++ b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
@@ -196,6 +196,9 @@ declare_args() {
     rtc_use_h265 = proprietary_codecs
   }
 
+  # Allow to use system OpenH264
+  rtc_use_system_openh264 = false
+
   # Enable this flag to make webrtc::Mutex be implemented by absl::Mutex.
   rtc_use_absl_mutex = false
 
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 9b9be814d..f4dab69d5 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -449,6 +449,10 @@ foreach(arch ${archs})
                 use_bluez=false
                 is_cfi=false
                 use_ozone=true
+                media_use_openh264=true
+                rtc_use_h264=true
+                rtc_use_system_openh264=true
+                rtc_system_openh264=true
                 ozone_auto_platforms=false
                 ozone_platform_headless=false
                 ozone_platform_external=true
