Submodule src/3rdparty e4b285617..8e38c6d31:
diff --git a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
index 66ed8434ff05..1bc73070d395 100644
--- a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
+++ b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
@@ -196,6 +196,9 @@ declare_args() {
     rtc_use_h265 = proprietary_codecs
   }
 
+  # Allow to use system OpenH264
+  rtc_use_system_openh264 = false
+
   # Enable this flag to make webrtc::Mutex be implemented by absl::Mutex.
   rtc_use_absl_mutex = false
 
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index f65567cc7..f5c914179 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -451,6 +451,10 @@ foreach(arch ${archs})
                 libyuv_use_sme=false # FIXME: arch not supported on our old CI gcc.
                 libyuv_use_neon=false # FIXME: udot instruction not working on our old CI gcc.
                 use_ozone=true
+                media_use_openh264=true
+                rtc_use_h264=true
+                rtc_use_system_openh264=true
+                rtc_system_openh264=true
                 ozone_auto_platforms=false
                 ozone_platform_headless=false
                 ozone_platform_external=true
