Submodule src/3rdparty 8e38c6d31..05cd29148:
diff --git a/src/3rdparty/chromium/third_party/blink/renderer/modules/mediarecorder/BUILD.gn b/src/3rdparty/chromium/third_party/blink/renderer/modules/mediarecorder/BUILD.gn
index f30411eddcfb..277b440c9ea6 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/modules/mediarecorder/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/blink/renderer/modules/mediarecorder/BUILD.gn
@@ -3,9 +3,19 @@
 # found in the LICENSE file.
 
 import("//build/buildflag_header.gni")
+import("//build/config/linux/pkg_config.gni")
 import("//media/media_options.gni")
 import("//third_party/blink/renderer/modules/modules.gni")
 import("//third_party/libaom/options.gni")
+import("//third_party/webrtc/webrtc.gni")
+
+if (rtc_use_h264 && rtc_use_system_openh264) {
+  pkg_config("openh264") {
+    packages = [
+      "openh264",
+    ]
+  }
+}
 
 blink_modules_sources("mediarecorder") {
   sources = [
@@ -57,7 +67,11 @@ blink_modules_sources("mediarecorder") {
       "h264_encoder.h",
     ]
 
-    deps += [ "//third_party/openh264:encoder" ]
+    if (rtc_use_system_openh264) {
+      configs += [":openh264"]
+    } else {
+      deps += [ "//third_party/openh264:encoder" ]
+    }
   }
 
   if (enable_libaom) {
diff --git a/src/3rdparty/chromium/third_party/blink/renderer/modules/webcodecs/BUILD.gn b/src/3rdparty/chromium/third_party/blink/renderer/modules/webcodecs/BUILD.gn
index 2f870757803d..1439b0f2f248 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/modules/webcodecs/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/blink/renderer/modules/webcodecs/BUILD.gn
@@ -2,12 +2,22 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//build/config/linux/pkg_config.gni")
 import("//media/media_options.gni")
 import("//third_party/blink/renderer/modules/modules.gni")
 import("//third_party/libaom/options.gni")
 import("//third_party/libgav1/options.gni")
 import("//third_party/libprotobuf-mutator/fuzzable_proto_library.gni")
 import("//third_party/protobuf/proto_library.gni")
+import("//third_party/webrtc/webrtc.gni")
+
+if (rtc_use_h264 && rtc_use_system_openh264) {
+  pkg_config("openh264") {
+    packages = [
+      "openh264",
+    ]
+  }
+}
 
 blink_modules_sources("webcodecs") {
   sources = [
@@ -122,7 +132,11 @@ blink_modules_sources("webcodecs") {
     deps += [ "//third_party/libvpx" ]
   }
   if (media_use_openh264) {
-    deps += [ "//third_party/openh264:encoder" ]
+    if (rtc_use_system_openh264) {
+      configs += [":openh264"]
+    } else {
+      deps += [ "//third_party/openh264:encoder" ]
+    }
   }
   if (is_fuchsia) {
     deps += [ "//media/fuchsia/video" ]
