diff --git a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
index 5a1c43c8888..d867f7e5330 100644
--- a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
+++ b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
@@ -186,11 +186,10 @@ declare_args() {
   #
   # Enabling H264 when building with MSVC is currently not supported, see
   # bugs.webrtc.org/9213#c13 for more info.
-  rtc_use_h264 =
-      proprietary_codecs && !is_android && !is_ios && !(is_win && !is_clang)
+  rtc_use_h264 = false

   # Enable to use H265
-  rtc_use_h265 = proprietary_codecs
+  rtc_use_h265 = false

   # Enable this flag to make webrtc::Mutex be implemented by absl::Mutex.
   rtc_use_absl_mutex = false
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index e7432d0c7..ea43a3f40 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -350,6 +350,10 @@ foreach(arch ${archs})
             use_static_angle=true
             use_perfetto_client_library=false
             trial_comparison_cert_verifier_supported=false
+            media_use_ffmpeg=true
+            media_use_openh264=false
+            rtc_use_h264=false
+
         )
         extend_gn_list(gnArgArg
             ARGS use_v8_context_snapshot v8_use_external_startup_data
